src/js/client/editprefset-mineAllSwitch-widget.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Mine / All Switch Widgets
+ *
+ * This includes a base grade and specific grades for a UI control that
+ * allows for switching between two states: Viewing all of the settings
+ * for a specific product/generic, or viewing only the settings that the
+ * user currently has personalized values for.  This allows a shorter list
+ * on the screen in the event that the category has tens or hundreds of
+ * settings.
+ *
+ * We will provide 2 implementations.  One that uses 2 buttons, and another
+ * that uses the Foundation boolean toggle. Additionally, the number of all
+ * or mine settings is includes in the control display.
+ */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.mineAllSwitch");
+
+fluid.defaults("gpii.devpmt.mineAllSwitch.base", {
+    gradeNames: ["gpii.devpmt.viewComponent"],
+    model: {
+        settingsFilter: "allsettings",
+        allSettingsCount: 0,
+        mySettingsCount: 0
+    },
+    invokers: {
+        // Overriding renderInitialMarkup to use `that` as a render context.
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
```

Comment:
Is the definition of `renderMarkup()` function originated from [infusion](https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/NewViewSupport.js#L64)? If it is, why there are so many args provided at line 32?

Reply:
This is being overridden in order to pass `that` as the render context, whereas typically the `model` is being passed in ( I believe this is from gpii-handlebars actually ). It's the 3rd argument, so we are providing at least up to that argument.